cli/cmd/root.go

```diff
@@ -137,6 +129,11 @@ func NewCmdRoot() *cobra.Command {
 	)
 	cmd.CompletionOptions.HiddenDefaultCmd = true
 	cmd.DisableAutoGenTag = true
+	cobra.OnFinalize(func() {
+		if analyticsClient != nil {
```

Comment:
This is the way to clean up resources with Cobra. However we can't close the log file here since we use it later to print the error

Reply:
Maybe it could be done with `runtime.SetFinalizer` but it would be hacky at best. There's a fairly new [runtime.addExitHook](https://go-review.googlesource.com/c/go/+/354790) discussion/patch which is merged but since it looks to be unexported I think it's not meant for outside use yet.